config/file_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func Test_LookupAuthConfig_WithNoConfigFile(t *testing.T) {
+	DefaultDir, _ = ioutil.TempDir("", "faas-cli-file-test")
+	DefaultFile = "test1.yml"
+	_, _, err := LookupAuthConfig("http://openfaas.test1")
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:config file not found)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
```

Comment:
I don't like the pattern we have in these tests (it's not from Stefan, but from some historic tests that got copied) `Error not matched` is super vague.. can we do better? What would you suggest guys? @stefanprodan @johnmccabe 

Imagine reading this deep in an CI log.

Reply:
I tend to use table drive as much as possible with the name of the testing being printed and for assertion failures printing the both the expected and actual values, kinda like:

    t.Errorf("Returned error [%v] does not match expected value [%v]", err, expectedErr)

@stefanprodan as an aside, you should be using `%v` with the raw error rather than `%s` and calling the `.String()` function.